### PR TITLE
Feature/cld 419

### PR DESCRIPTION
**Pull request by twosdai**

### CODE REVIEW

**Review by vitalykorolev — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -165,6 +165,27 @@ MarkLogic Server Docker containers are configured using a set of environment var
...
+|OVERWRITE_ML_CONF   | true                            | no                                | false     | Deletes and rewrites `/etc/marklogic.conf` with the passed in env variables if set to true | 
+
+
+MarkLogic Server also can be configured through a configuration file on the image to `/etc/marklogic.conf`. To interface with the configuration file pass in `OVERWRITE_ML_CONF` set to `true` and the following env variables can be written, if set, to `/etc/marklogic.conf`. 
+
+
+ | env var                       | value                           | required                          | default   | description                                        |
+| ------------------------------- | --------------------------------- | ----------------------------------- | ----------- | ---------------------------------------------------- |
+| MARKLOGIC_USER                | daniel                            | no                                |     n/a      | The username running MarkLogic within the docker container           |
```
Comment (jkerr5) [inline README.md:176]: I would move MARKLOGIC_USER, MARKLOGIC_DISABLE_JVM, JAVA_HOME, CLASSPATH to the bottom of the list and move ML_HUGEPAGES_TOTAL above them. Those settings are more "advanced" and would only be used if building a new image with this as the base.

**Review by jkerr5 — CHANGES_REQUESTED** [review]

File: src/centos/scripts/start-marklogic.sh [hunk 2]
```diff
@@ -13,6 +13,38 @@
...
+    echo "Deleting previous /etc/marklogic.conf ,if it exists, and overwriting with env variables"
+    rm -f /etc/marklogic.conf
+    sudo touch /etc/marklogic.conf && sudo chmod 777 /etc/marklogic.conf
+
+    [[ $MARKLOGIC_USER ]] && sudo echo "export MARKLOGIC_USER=$MARKLOGIC_USER" >> /etc/marklogic.conf
+    [[ $MARKLOGIC_PID_FILE ]] && sudo echo "export MARKLOGIC_PID_FILE=$MARKLOGIC_PID_FILE" >> /etc/marklogic.conf
+    [[ $MARKLOGIC_UMASK ]] && sudo echo "export MARKLOGIC_UMASK=$MARKLOGIC_UMASK" >> /etc/marklogic.conf
+    [[ $MARKLOGIC_DISABLE_JVM ]] && sudo echo "export MARKLOGIC_DISABLE_JVM=$MARKLOGIC_DISABLE_JVM" >> /etc/marklogic.conf
+    [[ $TZ ]] && sudo echo "export  TZ=$TZ " >> /etc/marklogic.conf
+    [[ $MARKLOGIC_NODE_NAME ]] && sudo echo "export MARKLOGIC_NODE_NAME=$MARKLOGIC_NODE_NAME" >> /etc/marklogic.conf
```
[the remark this answers is not in the document]
Comment (twosdai) [reply src/centos/scripts/start-marklogic.sh]: Updated: 2252330a5ab8bf8b4ff962e890da90b8111b7f35

File: src/centos/scripts/start-marklogic.sh [hunk 3]
```diff
@@ -13,6 +13,38 @@
...
+    [[ $MARKLOGIC_ADMIN_USERNAME ]] && sudo echo "export MARKLOGIC_ADMIN_USERNAME=$MARKLOGIC_ADMIN_USERNAME" >> /etc/marklogic.conf
+    [[ $MARKLOGIC_ADMIN_PASSWORD ]] && sudo echo "export MARKLOGIC_ADMIN_PASSWORD=$MARKLOGIC_ADMIN_PASSWORD" >> /etc/marklogic.conf
+    [[ $MARKLOGIC_LICENSEE ]] && sudo echo "export MARKLOGIC_LICENSEE=$MARKLOGIC_LICENSEE" >> /etc/marklogic.conf
+    [[ $MARKLOGIC_LICENSE_KEY ]] && sudo echo "export MARKLOGIC_LICENSE_KEY=$MARKLOGIC_LICENSE_KEY" >> /etc/marklogic.conf
+    [[ $MARKLOGIC_DISABLE_JVM ]] && sudo echo "export MARKLOGIC_DISABLE_JVM=$MARKLOGIC_DISABLE_JVM" >> /etc/marklogic.conf
```
[the remark this answers is not in the document]
Comment (twosdai) [reply src/centos/scripts/start-marklogic.sh:37]: Updated: 2252330a5ab8bf8b4ff962e890da90b8111b7f35

File: src/centos/scripts/start-marklogic.sh [hunk 4]
```diff
@@ -13,6 +13,38 @@
...
+    [[ $JAVA_HOME ]] && sudo echo "export JAVA_HOME=$JAVA_HOME" >> /etc/marklogic.conf
+    [[ $CLASSPATH ]] && sudo echo "export CLASSPATH=$CLASSPATH" >> /etc/marklogic.conf
```
[the remark this answers is not in the document]
Comment (twosdai) [reply src/centos/scripts/start-marklogic.sh:40]: Updated: 2252330a5ab8bf8b4ff962e890da90b8111b7f35

File: README.md (hunk 1 above)
Comment (twosdai) [reply README.md]: See: https://github.com/marklogic/marklogic-docker/commit/be2b425cb7ccc9d7a4fde23dd06317b534b48c82 for the new format. LMK what you think, (copy from the previous comment on the env variables)

**Review by jkerr5 — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -165,6 +165,36 @@ MarkLogic Server Docker containers are configured using a set of environment var
 | LICENSE_KEY           | license key                     | no                                | n/a       | set MarkLogic license key                          |
 | LICENSEE            | licensee information            | no                                | n/a       | set MarkLogic licensee information                 |
 |INSTALL_CONVERTERS   | true                            | no                                | false     | Installs converters for the client if they are not already installed | 
+|OVERWRITE_ML_CONF   | true                            | no                                | false     | Deletes and rewrites `/etc/marklogic.conf` with the passed in env variables if set to true | 
+
+
+MarkLogic Server also can be configured through a configuration file on the image to `/etc/marklogic.conf`. To interface with the configuration file pass in `OVERWRITE_ML_CONF` set to `true` and the following env variables can be written, if set, to `/etc/marklogic.conf`.
```
Comment (jmckean17) [inline README.md:171]: "MarkLogic Server also can be configured through a configuration file on the image at...

"To change the configuration file, pass in the parameter.... set to 'true. The following env variables can also be written to the '/etc/marklogic.conf' if the parameter is set.

File: README.md [hunk 6]
```diff
@@ -165,6 +165,36 @@ MarkLogic Server Docker containers are configured using a set of environment var
...
+| MARKLOGIC_LICENSEE      | licensee information                         | no | n/a       | set MarkLogic licensee information                           |
+| MARKLOGIC_LICENSE_KEY                | license key                             | no                                |   n/a        | set MarkLogic license key             |
+| ML_HUGEPAGES_TOTAL      | 1000                        | no | n/a       | set the number of huge pages marklogic can utilize                          |
+
+
+
+
+
+#### Advanced Configuration 
+The following environment variables are only useful when building and extending the current docker image. For instance setting `MARKLOGIC_USER` only will work if a user is setup and configured in the image.
```
Comment (jmckean17) [inline README.md:188]: For instance,
(comma)

set up
(two words)

File: README.md [hunk 7]
```diff
@@ -165,6 +165,36 @@ MarkLogic Server Docker containers are configured using a set of environment var
...
+| ------------------------------- | --------------------------------- | ----------------------------------- | ----------- | ---------------------------------------------------- |
+| MARKLOGIC_USER                | daniel                            | no                                |     n/a      | The username running MarkLogic within the docker container           |
+| MARKLOGIC_DISABLE_JVM      | 0                        | no | n/a       | disable the JVM for MarkLogic
+| JAVA_HOME                | /var/opt/java                            | no                                |  n/a         | set the java home location for MarkLogic           |
+| CLASSPATH      | /var/opt/class/path                        | no| n/a       | set the java env class path                          |
+| MARKLOGIC_PID_FILE      | /var/run/MarkLogic.pid                        | no| n/a       | The process ID file                         |
+| MARKLOGIC_UMASK      | 022                        | no | n/a       | The permissions granted to marklogic through umask                          |
```
Comment (jmckean17) [inline README.md:197]: capitalize MarkLogic in the description?

File: src/centos/scripts/start-marklogic.sh [hunk 8]
```diff
@@ -13,6 +13,38 @@
 
 cd ~
 
+################################################################
+# Read in ENV values
+################################################################
+
+# If an ENV value exists in a list Append it the /etc/marklogic.conf file
```
Comment (jmckean17) [inline src/centos/scripts/start-marklogic.sh:20]: ...list, append it to the ....

**Review by jmckean17 — COMMENTED** [review]
a bit of wordsmithing

File: README.md (hunk 6 above)
Comment (twosdai) [reply README.md]: Updated 8850cb5f8497deddba22339da2962ad3a1184a69

File: README.md (hunk 7 above)
Comment (twosdai) [reply README.md]: 8850cb5f8497deddba22339da2962ad3a1184a69 Updated

File: src/centos/scripts/start-marklogic.sh (hunk 8 above)
Comment (twosdai) [reply src/centos/scripts/start-marklogic.sh]: 8850cb5f8497deddba22339da2962ad3a1184a69 updated

File: README.md (hunk 5 above)
Comment (twosdai) [reply README.md]: 09912a67e64860f79171c63e9bc7e7e432f8d02b Updated

**Review by barkhachoithani — APPROVED** [review]
looks good to me

**Pull request merged**